internal/image/validate.go

```diff
@@ -130,7 +131,7 @@ func ValidateImage(ctx context.Context, comp app.SnapshotComponent, snap *app.Sn
 			return nil, err
 		}
 		allResults = append(allResults, results...)
-		out.Data = append(out.Data, data)
+		out.Data[fmt.Sprintf("%d", idx)] = append(out.Data[fmt.Sprintf("%d", idx)], data)
```

Comment:
Could be a tad nicer
```suggestion
		key := strconv.FormatInt(idx, 10)
		out.Data[key] = append(out.Data[key], data)
```